### PR TITLE
ENH: Warn on unsupported Python 3.10+

### LINKED ISSUE #17349: update PyPI classifiers to not install with unsupported Python versions
Right now if you do `pip-3.9 install numpy`, it will try to build NumPy for a version that does not support Python 3.9 yet (as stated in the release notes, and in `numpy/__init__.py`). We should see if we can fix the PyPI classifiers for this.

EDIT: or it's `python_requires='>=3.6'` in `setup.py`.

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/numpy/numpy/issues/17349.

First commit (https://github.com/numpy/numpy/pull/17443/commits/f3a6b33144e62ce77aa767b73d930dec420a3511): warn when installing on unsupported (new) Python versions.

Based on the similar warning we have in Pillow:

* https://github.com/python-pillow/Pillow/blob/66209168847ad1b55190a629b49cc6ba829efe92/setup.py#L40-L47

(Not sure if this is an ENT or MAINT.)

---

Second commit (https://github.com/numpy/numpy/pull/17443/commits/2ebb45374d845ad6d4843e977c0ca03e53674d25): whilst editing `setup.py`, I also fixed some Flake8 warnings in there. Let me know if I should drop this commit, or parts of it.

### CODE REVIEW

**Comment by charris** [comment]
Python 3.9 will be released tomorrow and numpy 1.19.3+ will will support it.

**Comment by hugovk** [comment]
Shall I bump it to 3.10+?

File: setup.py [hunk 1]
```diff
@@ -59,6 +59,13 @@
 ISRELEASED          = False
 VERSION             = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
 
+if sys.version_info >= (3, 9):
```
Comment (eric-wieser) [inline setup.py:62]:
```suggestion
# keep this consistent with the `Programming Language :: Python :: ...` classifiers above
if sys.version_info >= (3, 9):
```

File: setup.py [hunk 2]
```diff
@@ -59,6 +59,13 @@
...
+    warnings.warn(
+        f"NumPy {VERSION} does not support Python "
```
Comment (eric-wieser) [inline setup.py:64]:
```suggestion
        f"NumPy {VERSION} does yet not support Python "
```
or perhaps
```suggestion
        f"NumPy {VERSION} may not yet support Python "
```

**Comment by charris** [comment]
> Shall I bump it to 3.10+?

Yes. 1.19.x already builds with 3.9, we are just waiting for 3.9 to show up on the various platforms for which we produce binary wheels. In general, we try to have a release out within a week or two of the Python release and have working code for the late betas. We can probably update the version when we start testing against it.

File: setup.py [hunk 3]
```diff
@@ -59,6 +59,14 @@
 ISRELEASED          = False
 VERSION             = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
 
+# keep this consistent with the `Programming Language :: Python :: ...` classifiers above
+if sys.version_info >= (3, 10):
```
Comment (eric-wieser) [inline setup.py:63]: By changing this to 3.10, you've already made this inconsistent!

Comment (hugovk) [reply setup.py]: Ha, good point! Updated, I can rebase and squash these commit when we're done.

**Review by eric-wieser — APPROVED** [review]

**Comment by charris** [comment]
Thanks @hugovk .

**Pull request merged**